tests/unittests/CoreText/CTFontTests.mm

```diff
@@ -88,4 +88,97 @@ virtual void TearDown() {
+    CTFontRef font3 = CTFontCreateWithName(CFSTR("Times New Roman Bold"), 12.0, NULL);
+    CTFontRef font4 = CTFontCreateWithName(CFSTR("Times New Roman"), 14.0, NULL);
+    CFAutorelease(font1);
+    CFAutorelease(font2);
+    CFAutorelease(font3);
+    CFAutorelease(font4);
+
+    EXPECT_TRUE(CFEqual(font1, font2));
+    EXPECT_FALSE(CFEqual(font1, font3));
+    EXPECT_FALSE(CFEqual(font1, font4));
+
+    EXPECT_EQ(CFHash(font1), CFHash(font2));
+    EXPECT_NE(CFHash(font1), CFHash(font3));
+    EXPECT_NE(CFHash(font1), CFHash(font4));
+}
+
+TEST(CTFont, Description) {
+    CTFontRef font1 = CTFontCreateWithName(CFSTR("Arial"), 10.0, NULL);
+    CFAutorelease(font1);
+
+    EXPECT_OBJCNE(nil, (id)CFAutorelease(CFCopyDescription(font1)));
+}
+
+TEST(CTFont, CopyNameHelpers) {
+    CTFontRef font = CTFontCreateWithName(CFSTR("Arial Italic"), 15.0, NULL);
+    CFAutorelease(font);
+
+    EXPECT_OBJCEQ(@"Arial-ItalicMT", (id)CFAutorelease(CTFontCopyPostScriptName(font)));
+    EXPECT_OBJCEQ(@"Arial", (id)CFAutorelease(CTFontCopyFamilyName(font)));
+    EXPECT_OBJCEQ(@"Arial Italic", (id)CFAutorelease(CTFontCopyFullName(font)));
+    EXPECT_OBJCEQ(@"Arial Italic", (id)CFAutorelease(CTFontCopyDisplayName(font)));
+
+    font = CTFontCreateWithName(CFSTR("Arial Narrow"), 15.0, NULL);
+    CFAutorelease(font);
+
+    EXPECT_OBJCEQ(@"ArialNarrow", (id)CFAutorelease(CTFontCopyPostScriptName(font)));
+    EXPECT_OBJCEQ(@"Arial Narrow", (id)CFAutorelease(CTFontCopyFamilyName(font)));
+    EXPECT_OBJCEQ(@"Arial Narrow", (id)CFAutorelease(CTFontCopyFullName(font)));
+    EXPECT_OBJCEQ(@"Arial Narrow", (id)CFAutorelease(CTFontCopyDisplayName(font)));
+}
+
+TEST(CTFont, Metrics) {
+    CTFontRef font = CTFontCreateWithName(CFSTR("Courier New Bold Italic"), 15.0, NULL);
+    CFAutorelease(font);
+
+    EXPECT_OBJCEQ(@"Courier New Bold Italic", (id)CFAutorelease(CTFontCopyFullName(font)));
+
+    EXPECT_NE(0, CTFontGetAscent(font));
+    EXPECT_NE(0, CTFontGetDescent(font));
+    EXPECT_NE(0, CTFontGetUnitsPerEm(font));
+    EXPECT_NE(0, CTFontGetUnderlinePosition(font));
+    EXPECT_NE(0, CTFontGetUnderlineThickness(font));
+    EXPECT_NE(0, CTFontGetCapHeight(font));
+    EXPECT_NE(0, CTFontGetXHeight(font));
+}
+
+TEST(CTFont, GlyphCount) {
+    CTFontRef font = CTFontCreateWithName(CFSTR("Arial"), 15.0, NULL);
+    CFAutorelease(font);
+    // Test for reasonable bounds on number of unique glyphs in a font
+    EXPECT_LE(3000, CTFontGetGlyphCount(font));
+    EXPECT_GE(8000, CTFontGetGlyphCount(font));
+
+    font = CTFontCreateWithName(CFSTR("Courier New"), 9.0, NULL);
+    CFAutorelease(font);
+    // Test for reasonable bounds on number of unique glyphs in a font
+    EXPECT_LE(3000, CTFontGetGlyphCount(font));
+    EXPECT_GE(8000, CTFontGetGlyphCount(font));
+}
+
+TEST(CTFont, UnknownName) {
+    EXPECT_OBJCEQ(nil, (id)CFAutorelease(CTFontCreateWithName(CFSTR("DoesNotExistFont"), 12.0, NULL)));
+}
+
+// TODO: Do we care enough about fixing this?
```

Comment:
It was less 'cannot' and more 'this seems annoying to implement for very little gain'. Keeping the note here, then.

---
In reply to: [78618340](https://github.com/Microsoft/WinObjC/pull/987#discussion_r78618340) [](ancestors = 78618340,78609035)